pyam/core.py

```diff
@@ -261,34 +261,43 @@ def metadata(self, meta, name=None):
             (by `['model', 'scenario']` index if possible)
         name: str
             category column name (if not given by meta pd.Series.name)
+        index: pd.MultiIndex
+            index to be used for setting metadata column
         """
-        if isinstance(meta, pd.Series) and \
-                set(['model', 'scenario']).issubset(meta.index.names):
-            meta.name = name or meta.name
+        # create pd.Series from index and meta if provided
+        _meta = meta if index is None else pd.Series(data=meta, index=index)
+
```

Comment:
this may be a bit fancy, but I would suggest changes as follows:

```
try:
    append_by_idx = set(['model', 'scenario']).issubset(_meta.index.names)
except:
    append_by_idx = False

if not append_by_idx:
    if not name and not hasattr(meta, 'name'):
        raise ValueError('Must pass a name or use a pd.Series')
    name = name or _meta.name
    _meta = list(meta)
   self.meta[name] = meta

# then follow with the more complicated bit at normal indentation
```

Reply:
was a bit of an effort and not sure if this is necessarily prettier - but implemented